iidm/iidm-extensions/src/main/java/com/powsybl/iidm/network/extensions/GeneratorTanPhi.java

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.iidm.network.extensions;
+
+import com.powsybl.commons.extensions.Extension;
+import com.powsybl.iidm.network.Generator;
+
+/**
+ * @author Sophie Frasnedo {@literal <sophie.frasnedo at rte-france.com>}
+ */
+public interface GeneratorTanPhi extends Extension<Generator> {
```

Comment:
Not at all! I don't like that name either and I hope to find a better alternative.

Reply:
CIM defines `cim:RegulatingControlModeKind.powerFactor`, I would suggest `GeneratorPowerFactorRegulation`.

but beware that powerFactor is cosPhi, not tanPhi

Reply:
So there is no standard term for tanPhi? Any if modelling using cos phi instead of tan phi is the standard way we should also use it, this is equivalent.

Reply:
Ok thanks for your feedback @geofjamg and @jeandemanged !

We would define a powerFactor parameter and then do a `tan(acos(powerFactor))` to get what we need?
Should we allow the whole [-1; 1] range for cosPhi? In practice, it should always be close to 1, right?

I am not sure how the extension would be populated... By hand? We should think of a method to make it user-friendly for users who have the Q/P ratio in mind rather than the P/S one.